### ISSUE TITLE
False positive for PERF402 when using `.append()` on non-list objects

### ISSUE THREAD

**Issue by dosisod** [astral-sh/ruff#5580]
The following code emits an error despite it not being a list (or list-like):

```python
class C:
    def append(self, x: int) -> None:
        pass

c = C()

for line in [1, 2, 3]:
    c.append(line)
```

Running:

```
$ ruff file.py
file.py:8:5: PERF402 Use `list` or `list.copy` to create a copy of a list
Found 1 error.
```

Ruff version:

```
$ ruff --version
ruff 0.0.277
```

**Comment by charliermarsh** [comment]
In practice, can you say more about the class this is triggering on? Is it defined locally? Third-party? Etc.

**Comment by dosisod** [comment]
The error is on a class that I defined. The code that is throwing the error is [here](https://github.com/Cicada-Software/cicada/blob/b5ef1f2e5efc2183192c5a1e9a47af1ad8385f2d/test/api/test_terminal_session.py#L45-L53):

```python
from cicada.api.domain.terminal_session import TerminalSession

# snip

async def test_terminal_session_can_be_replayed_after_closing() -> None:
    session = TerminalSession()

    sent_lines = [b"line 1\n", b"line 2\n", b"line 3\n"]

    for line in sent_lines:
        session.append(line)  # noqa: PERF402

    session.finish()

    # snip
```

The code for the `TerminalSession` class is defined [here](https://github.com/Cicada-Software/cicada/blob/main/cicada/api/domain/terminal_session.py).

It could be argued that `.append()` name should be reserved for list-like objects (in my case `.write()` would probably be a better name), though [lots of code](https://grep.app/search?q=def%20append%28&case=true&filter[lang][0]=Python) defines their own `append()` class methods.

**Comment by charliermarsh** [comment]
This got fixed -- we're now more conservative and only enforce this on known lists.
